lib/widgets/topic_list.dart

```diff
@@ -0,0 +1,301 @@
+            topic: topic,
+          );
+        },
+      ),
+    );
+  }
+}
+
+class TopicListItem extends StatelessWidget {
+  const TopicListItem({
+    super.key,
+    required this.streamId,
+    required this.topic,
+  });
+
+  final int streamId;
+  final GetStreamTopicsEntry topic;
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final designVariables = DesignVariables.of(context);
+
+    final unreads = store.unreads.countInNarrow(
+      TopicNarrow(streamId, topic.name)
+    );
+    final hasUnreads = unreads > 0;
+
+    return Material(
+      type: MaterialType.transparency,
+      child: InkWell(
+        onTap: () {
+          Navigator.push(context, MessageListPage.buildRoute(
+            context: context,
+            narrow: TopicNarrow(streamId, topic.name),
+          ));
+        },
+        onLongPress: () {
+          showTopicActionSheet(
+            context,
+            channelId: streamId,
+            topic: topic.name,
+            someMessageIdInTopic: null,
+          );
+        },
+        child: Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
+          const SizedBox(width: 28),
+          Padding(
+            padding: const EdgeInsets.only(top: 10.0),
+            child: SizedBox(
+              height: 16,
+              width: 16,
+              child: topic.name.displayName.startsWith('✔')
+                ? Icon(
+                    ZulipIcons.check,
+                    size: 12,
+                    color: designVariables.textMessage,
+                  )
+                : null,
+            ),
+          ),
+          Expanded(
+            child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 8),
+              child: Row(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Expanded(
+                    child: Text(
+                      topic.name.displayName.startsWith('✔')
+                          ? topic.name.displayName.substring(2).trim()
+                          : topic.name.displayName,
+                      style: TextStyle(
+                        fontSize: 17,
+                        fontWeight: FontWeight.w400,
+                        height: (20 / 17),
+                        color: designVariables.textMessage,
+                        fontStyle: topic.name.displayName == 'general chat'
+                            ? FontStyle.italic
+                            : FontStyle.normal,
+                      ),
+                    ),
+                  ),
+                  const SizedBox(width: 8),
+                  if (hasUnreads) ...[
+                    UnreadCountBadge(
+                      count: unreads,
+                      backgroundColor: designVariables.bgCounterUnread,
+                      bold: true,
+                    ),
+                  ],
+                ],
```

Comment:
There is 12px padding before the end:

![image](https://github.com/user-attachments/assets/96e475ac-8b3f-4164-b523-357a8486db0c)